interactive_ai/workflows/otx_domain/trainer/otx_v2/Dockerfile.xpu

```diff
@@ -0,0 +1,53 @@
+FROM python:3.10-slim-bookworm AS base
+
+ENV UV_COMPILE_BYTECODE=1
+ENV UV_LINK_MODE=copy
+ENV UV_PYTHON_DOWNLOADS=0
+
+# Install runtime dependencies and intel graphics libs
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    libgl1=1.6.* \
+    libglib2.0-0=2.74.* \
+    software-properties-common \
+    gpg \
+    wget \
+    curl && \
+    rm -rf /var/lib/apt/lists/* && \
+    useradd -l -u 10001 non-root && \
+    pip3 uninstall -y setuptools pip wheel && \
+    rm -rf /root/.cache/pip
+
+RUN wget -qO /tmp/intel-graphics.key https://repositories.intel.com/gpu/intel-graphics.key && \
+    gpg --yes --dearmor --output /usr/share/keyrings/intel-graphics.gpg /tmp/intel-graphics.key
+RUN echo "deb [arch=amd64,i386 signed-by=/usr/share/keyrings/intel-graphics.gpg] https://repositories.intel.com/gpu/ubuntu jammy unified" | tee /etc/apt/sources.list.d/intel-gpu-jammy.list
+RUN apt-get update \
+    && apt-get install -y --no-install-recommends \
+    libze-intel-gpu1 libze1 intel-opencl-icd clinfo libze-dev intel-ocloc
+
+# Copy the service dependencies
+COPY --link --from=libs . libs
+COPY --link --from=ghcr.io/astral-sh/uv:0.6.12 /uv /bin/uv
+
+WORKDIR /interactive_ai/workflows/otx_domain/trainer/otx_v2
+
+COPY --link scripts/ scripts
+COPY --link run run
+COPY --link download_pretrained_weights.py download_pretrained_weights.py
+
+# For OTX to install torch with ipex support, else defaults to cpu version
+ENV PIP_EXTRA_INDEX_URL=https://download.pytorch.org/whl/xpu
+
+RUN --mount=type=cache,target=/root/.cache/uv \
+    --mount=type=bind,source=uv.lock,target=uv.lock \
+    --mount=type=bind,source=pyproject.toml,target=pyproject.toml \
+    uv sync --frozen --no-dev --no-editable
```

Comment:
uv sync creates a virtual environment in the WORKDIR folder and installs dependencies in it instead of using a system environment, resulting in two Python interpreters being present. To save space can we try [pointing uv to the system interpreter](https://docs.astral.sh/uv/concepts/projects/config/#project-environment-path)? 

Or using distroless as a runtime image and copy the environment with dependences and installed binaries?

Reply:
It is already reusing the system interpreter. uv will only download python when needed. 
<img width="1705" alt="image" src="https://github.com/user-attachments/assets/c71bd3b5-1848-4620-abb8-7ec16fd9192c" />

Reply:
Confirmed by inspecting the Docker image: the uv Python is just a symlink to the system one

```shell
non-root@f123fd9d7a0c:/interactive_ai/workflows/otx_domain/trainer/otx_v2$ which python
/interactive_ai/workflows/otx_domain/trainer/otx_v2/.venv/bin/python
non-root@f123fd9d7a0c:/interactive_ai/workflows/otx_domain/trainer/otx_v2$ ls -l .venv/bin/python
lrwxrwxrwx 1 root root 22 May 23 08:09 .venv/bin/python -> /usr/local/bin/python3
```